lib/helpers/add-one.js

```diff
@@ -26,8 +27,11 @@ function addOne (state, doc, prefix) {
 
   delete doc.hoodie
 
-  return internals.put(state, internals.addTimestamps(doc))
+  return validate(state, doc)
 
+  .then(function () {
+    return internals.put(state, internals.addTimestamps(doc))
```

Comment:
Btw, why isn't `internals` API like `internals.put(state, doc)`? 
Because `internals.addTimestamps` is accessible inside the definition of `internals.put` as `this.addTimestamps`. Am I wrong about this?